themes/default/templates/analysis.html

```diff
@@ -0,0 +1,251 @@
+        }
+        .table_options {
+            display: flex;
+            gap: 10px;
+        }
+
+        .scrollbox {
+            overflow: auto;
+            height: 200px;
+        }
+
+        :is(td,th):not(:last-child) {
+            border-right: 1px dotted #555;
+        }
+
+        td {
+            max-width: 10em;
+        }
+        td > span {
+            display: block;
+            overflow-x: auto;
+        }
+
+        table {
+            border-collapse: initial;
+        }
+
+        td:first-child, thead tr:nth-child(2) th:first-child  {
+            position: sticky;
+            left: 0;
+            background: white;
+        }
+
+
+        thead tr:last-child td {
+            border-bottom: 2px solid #ddd;
+            background: #eee;
+        }
+        .table_display {
+            max-height: 80vh;
+            overflow-y: auto;
+        }
+
+        .table_display thead {
+            position: sticky;
+            top: 0;
+            background: white;
+            z-index: 1;
+        }
+
+        tr:nth-child(odd) td{
+            background: #f9f9f9;
+        }
+
+        body:not(.not-secure-context) > #secure-context-warning,
+        body.not-secure-context >  #analysis {
+            display: none; 
+        }
+    </style>
+</head>
+
+<body class="container container-fluid">
+    <header>
+        <h1 data-localise="analysis.header"></h1>
+    </header>
+    <main id="secure-context-warning">
+        <div class="alert alert-danger">
+            <p data-localise="analysis.not secure context"></p>
+        </div>
+    </main>
+    <main id="analysis">
+        <section data-bind="visible:current_tab()=='list_files'">
+
+            <h2 data-localise="analysis.upload files"></h2>
+            <div class="help-block">
+                <p data-localise="analysis.help.upload files"></p>
+            </div>
+
+            <form data-bind="submit: function() {}">
+                <p>
+                    <label for="upload_files" data-localise="analysis.help.file input label"></label>
+                    <input class="form-control" type="file" id="upload_files" accept=".txt" multiple data-bind="event: {change: input_files}">
+                </p>
+            </form>
+
+            <!-- ko if: uploaded_files().length -->
+
+            <p data-bind="text: file_summary"></p>
+
+            <table id="uploaded-files" class="table">
+                <thead>
+                    <tr>
+                        <th data-localise="exam.student name"></th>
+                        <th data-localise="analysis.start time"></th>
+                        <th data-localise="analysis.file"></th>
+                        <th></th>
+                    </tr>
+                </thead>
+                <tbody data-bind="foreach: sorted_files">
+                    <tr data-bind="class: status_class">
+                        <!-- ko if: !content() -->
+                        <td colspan="2">
```

Comment:
If valid and invalid files are both uploaded, the table does not line up - the error lines are 5 columns long whilst the processed lines are 4 columns long, download and delete being out-of-alignment.